torchdata/datapipes/iter/util/decompressor.py

```diff
@@ -87,6 +92,8 @@ def _detect_compression_type(self, path: str) -> CompressionType:
             return self.types.ZIP
         elif ext == ".bz2":
             return self.types.BZIP2
+        elif ext == ".zlib":
```

Comment:
There doesn't seem to be a strong standard for what extension zlib files should have. I've also seen .z and .zz.